### ISSUE TITLE
Silent Renew not working in new tab in Version 11.0.0 like old versions

### ISSUE THREAD

**Issue by Yajmir** [damienbod/angular-auth-oidc-client#718]
**Describe the bug**
In version 10.0.13 silent renew would kick in when you copy paste a URL into new tab. This didn't work in 10.0.14. I have updated to 11.0.0 and it also doesn't work in this version.

**To Reproduce**
  stsServer: 'http://localhost:5000',
          redirectUrl: 'http://localhost:4200/#/Dashboard',
          clientId: 'js',
          scope: 'openid profile web_api ',
          responseType: 'code',
          triggerAuthorizationResultEvent: true,
          postLogoutRedirectUri: 'http://localhost:4200',
          startCheckSession: true,
          silentRenew: true,
          silentRenewUrl: 'http://localhost:4200/silent-renew.html',
          postLoginRoute: '/#/doctor',
          forbiddenRoute: '/#/login',
          unauthorizedRoute: '/unauthorized',
          logLevel: LogLevel.Debug,

**Expected behavior**
Silent renew should kick in, in new tab.

**Comment by damienbod** [comment]
Hi @Yajmir Thanks for reporting. 

This is because the app uses sessionStorage per default and so the apps don't share this data between the tabs. So when you open the app in a new tab, you are not logged in. I see 2 ways you could solve this, one is to use `localStorage` instead of sessionStorage but this might make you app less safe. A second way is to do an Auto-Login when the APP opens up, if the APP is not doing a callback. We have an example for this in the samples.

Hope this helps

Greetings Damien

**Comment by Yajmir** [comment]
Thanks for the reply. This silent renew worked in 10.0.13. In that version auto login wouldn’t work. So I upgraded to 11.0.0. I have setup auto login and it works in this version. But the redirection doesn’t feel nice.

**Comment by damienbod** [comment]
yes we removed the default auto run silent renew and we had problems with the startup in version 10, so we refactored this away, but with breaking changes. The redirection is the only way I see at the moment to do this, maybe not. The problem is that only static known URLs on the STS can be used for redirect and so these have to be pre-registered. This is important for security.

A possible enhancement, would be to allow the URL redirect to be configured in the authorize request as an optional parameter. If you had a finite number of URLs in the APP, these could be added to the STS as valid redirects, and then the redirect logic we use could be completly replaced then. This would also be a bit messy I think.

Thanks for your feedback.

Greetings Damien
